applications/crossbar/test/cb_token_restrictions_test.hrl

```diff
@@ -0,0 +1,129 @@
+-define(ACCOUNT_DESCENDANT, <<"{DESCENDANT_ACCOUNT_ID}">>).
+
+-define(ACCOUNTS_AUTH, [?ACCOUNT_AUTH]).
+-define(ACCOUNTS_DESCENDANT, [?ACCOUNT_DESCENDANT]).
+-define(ACCOUNTS_ANY, [?CATCH_ALL]).
+-define(ACCOUNTS_EXACT, [?ACCOUNT_ID]).
+
+-define(ENDPOINT_DEVICE
+        ,wh_json:from_list(
+           [{<<"allowed_accounts">>, ?ACCOUNTS_AUTH}
+            ,{<<"rules">>, wh_json:new()}
+           ]
+          )
+       ).
+
+-define(DEVICE_RULES
+        ,wh_json:from_list([{<<"devices">>, [?ENDPOINT_DEVICE]}])
+       ).
+
+-define(ALLOW_ALL_RULE_RESTRICTIONS
+        ,wh_json:from_list(
+           [{?CATCH_ALL
+             ,[wh_json:from_list(
+                 [{<<"allowed_accounts">>, ?ACCOUNTS_ANY}
+                  ,{<<"rules">>, wh_json:from_list([{?ARGS_ANY, ?HTTP_ANY}])}
+                 ]
+                )
+              ]
+            }
+           ]
+          )
+       ).
+
+-define(DENY_API_ENDPOINT_RESTRICTIONS
+        ,wh_json:from_list(
+           [{<<"accounts">>
+             ,[wh_json:from_list(
+                 [{<<"allowed_accounts">>, ?ACCOUNTS_ANY}
+                  ,{<<"rules">>, wh_json:from_list([{?ARGS_ANY, ?HTTP_ANY}])}
+                 ]
+                )
+              ]
+            }
+           ]
+          )
+       ).
+
+-define(ALLOW_API_ENDPOINT_RESTRICTIONS
+        ,wh_json:from_list(
+           [{<<"devices">>
+             ,[wh_json:from_list(
+                 [{<<"allowed_accounts">>, ?ACCOUNTS_ANY}
+                  ,{<<"rules">>, wh_json:from_list([{?ARGS_ANY, ?HTTP_ANY}])}
+                 ]
+                )
+              ]
+            }
+           ]
+          )
+       ).
+
+-define(ALLOW_ACCOUNTS_RESTRICTIONS(AccountId)
+        ,wh_json:from_list(
+           [{<<"devices">>
+             ,[wh_json:from_list(
+                 [{<<"allowed_accounts">>
+                   ,case AccountId of 'undefined' -> 'undefined'; _ -> [AccountId] end
+                  }
+                  ,{<<"rules">>, wh_json:from_list([{?ARGS_ANY, ?HTTP_ANY}])}
+                 ]
+                )
+              ]
+            }
+           ]
+          )
+       ).
+
+-define(ARGUMENTS_RESTRICTIONS(Arg)
+        ,wh_json:from_list(
+           [{<<"devices">>
+             ,[wh_json:from_list(
+                 [{<<"allowed_accounts">>, [?ACCOUNT_ID]}
+                  ,{<<"rules">>, wh_json:from_list([{Arg, ?HTTP_ANY}])}
+                 ]
+                )
+              ]
+            }
+           ]
+          )
+       ).
+
+-define(HTTP_VERB_RESTRICTIONS(Verbs)
+        ,wh_json:from_list(
+           [{<<"devices">>
+             ,[wh_json:from_list(
+                 [{<<"allowed_accounts">>, [?ACCOUNT_ID]}
+                  ,{<<"rules">>, wh_json:from_list([{?ARGS_ANY, Verbs}])}
+                 ]
+                )
+              ]
+            }
+           ]
+          )
+       ).
+
+-define(CB_TOKEN_RESTRICTIONS_TEST_HRL, 'true').
+-endif.
```

Comment:
I put these macros in a hrl since they were big and clunky and cluttered the test module up. Easier to read `?HTTP_VERB_RESTRICTIONS(Verbs)` than the `wh_json:from_list()` monstrosity. If the macro was doing more than constructing a term, I would create a function but in this case it felt fine to put a "static" term builder in a macro instead.